src/libcore/fmt/mod.rs

```diff
@@ -1124,6 +1124,39 @@ impl<T: Debug> Debug for [T] {
     }
 }
 
+macro_rules! fmt_dst {
+    ($($Trait:ident => $fmt_char:expr),*) => {
+        $(
+            impl<T: $Trait> $Trait for [T] {
+                fn fmt(&self, f: &mut Formatter) -> Result {
+                    try!(write!(f, "["));
+                    let mut is_first = true;
+                    for x in self.iter() {
+                        if is_first {
```

Comment:
Different style here than from everything else. (`for (i, x) in iter.enumerate() { if i != 0 { .. } ..`)

Intentional?